src/Facebook/FacebookClient.php

```diff
@@ -88,43 +88,33 @@ class FacebookClient
     /**
      * Instantiates a new FacebookClient object.
      *
-     * @param FacebookHttpClientInterface|null $httpClientHandler
-     * @param boolean                          $enableBeta
+     * @param HttpClient|null $httpClient
+     * @param boolean         $enableBeta
      */
-    public function __construct(FacebookHttpClientInterface $httpClientHandler = null, $enableBeta = false)
+    public function __construct(HttpClient $httpClient = null, $enableBeta = false)
     {
-        $this->httpClientHandler = $httpClientHandler ?: $this->detectHttpClientHandler();
+        $this->httpClient = $httpClient ?: HttpClientDiscovery::find();
         $this->enableBetaMode = $enableBeta;
     }
 
     /**
      * Sets the HTTP client handler.
      *
-     * @param FacebookHttpClientInterface $httpClientHandler
+     * @param HttpClient $httpClient
      */
-    public function setHttpClientHandler(FacebookHttpClientInterface $httpClientHandler)
+    public function setHttpClient(HttpClient $httpClient)
     {
-        $this->httpClientHandler = $httpClientHandler;
+        $this->httpClient = $httpClient;
     }
 
     /**
      * Returns the HTTP client handler.
      *
-     * @return FacebookHttpClientInterface
+     * @return HttpClient
      */
-    public function getHttpClientHandler()
+    public function getHttpClient()
```

Comment:
I agree with @Ocramius. Exposing internals seems to be a common practice, but for what? The smallest API the better, you can even add it back if it proves to be necessary, but as it is going to be a major release, API cleanups like this can happen and it's even related to HTTPlug here, so why not?